File: Python/BOJ/1788 피보나치 수의 확장.py
import sys
n = int(sys.stdin.readline())
F = [0]*(1000001)
p = 1000000000
F[1] = 1
for i in range(2,1000001):
    F[i] = (F[i-1] + F[i-2]) % p

if n % 2 == 0 and n < 0:
    print(-1)
elif n == 0:
    print(0)
else:
    print(1)
print(F[abs(n)])

File: Java/BOJ/1788 피보나치 수의 확장.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        long[] dp = new long[1000001];
        dp[1] = 1;
        for (int i = 2; i < 1000001; i++) {
        	dp[i] = (dp[i-1] + dp[i-2]) % 1000000000;
        }
        if (n % 2 == 0 && n < 0) {
        	System.out.println(-1);
        }
        else if (n == 0) {
        	System.out.println(0);
        }
        else {
        	System.out.println(1);
        }
        System.out.println(dp[Math.abs(n)]);
    }
}